ingestion/functions/parsing/japan/japan.py

```diff
@@ -43,23 +40,48 @@ def convert_location(raw_entry):
         if term and term != "Unspecified"]
     return {"query":  ", ".join(query_terms)}
 
-
 def detect_notes(raw_notes):
     if "notes" in raw_notes:
         return raw_notes["notes"]
     else:
         return None
 
-
 def convert_date(raw_date: str):
     """
     Convert raw date field into a value interpretable by the dataserver.
-
+​
     The date filtering API expects mm/dd/YYYYZ format.
     """
-    date = datetime.strptime(raw_date, "%Y-%m-%d")
+    date = datetime.strptime(raw_date["dateAnnounced"], "%Y-%m-%d")
     return date.strftime("%m/%d/%YZ")
 
+def convert_additionalSources(additional_sourceURL: str):
+    sources = []
+    if "sourceURL" in additional_sourceURL:
+      sources.append({"sourceUrl": additional_sourceURL["sourceURL"]})
+    if "deathSourceURL" in additional_sourceURL:
+      sources.append({"sourceUrl": additional_sourceURL["deathSourceURL"]})
+    return sources or None
+
+def convert_outcome(raw_outcome: str, raw_death_date: str):
+    if "patientStatus" in raw_outcome:
+        if raw_outcome["patientStatus"] == "Deceased":
+            death_date = datetime.strptime(raw_death_date["deceasedDate"], "%Y-%m-%d")
+            return {"name": "outcome",
+                    "dateRange":{
+                        "start": death_date.strftime("%m/%d/%YZ"),
+                        "end": death_date.strftime("%m/%d/%YZ")
+                        },
+                    "value": "Death"}
+        elif raw_outcome["patientStatus"] == "Discharged" or raw_outcome["patientStatus"] == "Recovered":
+            return {"name": "outcome",
+                    "value": "Recovered"}
+        else:
+            return {"name": "outcome",
+                    "value": "Unknown"}       
+    else:
+        return {"name": "outcome",
+                "value": "Unknown"}
```

Comment:
Can make this a little simpler to follow by deleting the first else (79-81) and removing the `else:` part of the second block (82).